game/src/main/kotlin/content/skill/summoning/Crafting.kts

```diff
@@ -0,0 +1,321 @@
+        "Transform-10" -> transformScrolls(player, enumIndex, 10)
+        "Transform-X" -> {
+            val total = intEntry("Enter amount:")
+            transformScrolls(player, enumIndex, total)
+        }
+        "Transform-All" -> transformScrolls(player, enumIndex, Int.MAX_VALUE)
+    }
+}
+
+/**
+ * Crafts summoning familiar pouches.
+ *
+ * @param player: The player doing the crafting
+ * @param enumIndex: The index in the "summoning_pouch_ids_*" enum of the pouch being crafted
+ * @param amount: The amount of pouches the player is attempting to craft
+ */
+fun infusePouches(player: Player, enumIndex: Int, amount: Int) {
+    val pouchItemId = enums.get("summoning_pouch_ids_1").getInt(enumIndex)
+    val pouchItem = Item(itemDefinitions.get(pouchItemId).stringId)
+
+    val shards = getShards(pouchItem)
+    val charms = getCharms(pouchItem)
+    val pouches = getPouches(pouchItem)
+    val tertiaries = getTertiaries(pouchItem)
+
+    val xpPerCraft: Double = pouchItem.def["infuse_experience"]
+    val maxCraftable = maxCraftable(player, shards, charms, pouches, tertiaries) ?: return
+    val amountToCraft = min(amount, maxCraftable)
+
+    player.interfaces.close("summoning_pouch_creation")
+
+    player.inventory.transaction {
+        remove(shards.id, shards.amount * amountToCraft)
+        remove(charms.id, charms.amount * amountToCraft)
+        remove(pouches.id, pouches.amount * amountToCraft)
+        tertiaries.forEach { tertiary -> remove(tertiary.id, tertiary.amount * amountToCraft) }
+        add(pouchItem.id, amountToCraft)
+    }
+
+    if (player.inventory.transaction.error == TransactionError.None) {
+        player.anim("summoning_infuse")
+        player.exp(Skill.Summoning, xpPerCraft * amountToCraft)
+    }
+}
+
+/**
+ * Crafts summoning scrolls.
+ *
+ * @param player: The player doing the crafting
+ * @param enumIndex: The index in the "summoning_scroll_ids_*" enum of the scroll being crafted
+ * @param amount: The amount of pouches the player is attempting to turn into scrolls
+ */
+fun transformScrolls(player: Player, enumIndex: Int, amount: Int) {
+    val scrollItemId = enums.get("summoning_scroll_ids_1").getInt(enumIndex)
+    val scrollItem = Item(itemDefinitions.get(scrollItemId).stringId)
+
+    val pouchId = enums.get("summoning_scroll_ids_2").getKey(scrollItemId)
+    val pouchItem = Item(itemDefinitions.get(pouchId).stringId)
+
+    val maxTransformAmount = player.inventory.count(pouchItem.id)
+    val amountToTransform = min(amount, maxTransformAmount)
+
+    val xpPerCraft: Double = scrollItem.def["transform_experience"]
+
+    player.inventory.transaction {
+        remove(pouchItem.id, amountToTransform)
+        add(scrollItem.id, amountToTransform * 10)
+    }
+
+    if (player.inventory.transaction.error == TransactionError.None) {
+        player.anim("summoning_infuse")
+        player.exp(Skill.Summoning, xpPerCraft * amountToTransform)
```

Comment:
```suggestion
        player.exp(Skill.Summoning, xpPerCraft * amountToTransform)
        player.closeMenu()
```

For some reason creating pouches closes the interface but infusing scrolls does not, both should close though